modules/core/src/main/java/org/apache/ignite/internal/processors/service/IgniteServiceProcessor.java

```diff
@@ -1421,6 +1444,18 @@ private Service copyAndInject(ServiceConfiguration cfg, ServiceContextImpl svcCt
         }
     }
 
+    /** @param cfg Lazy service configuration. */
+    private void unmarshalNodeFilterIfNeeded(LazyServiceConfiguration cfg) throws IgniteCheckedException {
```

Comment:
Suggestion: method name, some kind of `validateServiceConfiguration`. Which can throw an exception with predefined action description as string. Like 'deploying', 'joining' . Might be useful further, for validation of interceptor class etc.

Reply:
It seems that this method does unmarshalling directly, and should be part of more abstract validation methods - validateNode, etc as it is now

Reply:
`validateNode` may contain `validateServiceConfig`. Other methods aren't named `validate`

Reply:
But they already do deployment checks, including configuration checks.